aptiche/lib/views/home/home_controller.dart

```diff
@@ -4,45 +4,60 @@ import 'package:aptiche/utils/enums.dart';
 import 'package:get/get.dart';
 
 class HomeController extends GetxController {
+  /// Finds the [GraphQLService] using [Getx] dependency injection
   final GraphQLService _graphQL = Get.find();
+
+  /// A boolean which returns true if the user has chose to display upcoming
+  /// quizzes
   RxBool upcomingQuiz = true.obs;
-  CurrentState homeState = CurrentState.busy;
 
-  late List<Quiz> quizzes;
-  List<Quiz> upcomingQuizzes = <Quiz>[];
-  List<Quiz> pastQuizzes = <Quiz>[];
-  List<Quiz> desiredList = <Quiz>[];
+  /// An enum which stores the current state of the [HomeScreen]
+  Rx<CurrentState> homeState = CurrentState.busy.obs;
+
+  /// A list which stores all the active quizzes.
+  late List<Quiz> activeQuizzes;
+
+  /// A list whioch stores all the upcoming quizzes.
+  late List<Quiz> upcomingQuizzes;
+
+  /// A list which stores all the past quizzes.
+  late List<Quiz> pastQuizzes;
+
+  /// A list which stores either `activeQuizzes`, `upcomingQuizzes` or
+  /// `pastQuizzes` depending on what the user selects.
+  List<Quiz>? desiredList;
 
   @override
   void onInit() {
     getQuizzes();
     super.onInit();
   }
 
-  void assignList() {
+  /// The function assigns a list the desiredList depending on the user's
+  /// requirement.
+  void assignList() async {
+    homeState.value = CurrentState.busy;
     (upcomingQuiz.value)
         ? desiredList = upcomingQuizzes
         : desiredList = pastQuizzes;
+    homeState.value = CurrentState.ready;
     update();
   }
 
+  /// The function calls the getQuizzesByTime query from [GraphQLService] and
+  /// assigns the results to the respective quiz.
   Future<void> getQuizzes() async {
-    homeState = CurrentState.busy;
-    quizzes = await _graphQL.getQuizzes();
-    if ((upcomingQuizzes.length + pastQuizzes.length) != quizzes.length)
-      for (final Quiz quiz in quizzes) {
-        if (quiz.active) {
-          upcomingQuizzes.add(quiz);
-        }
-        if (!quiz.active) {
-          pastQuizzes.add(quiz);
-        }
-      }
-    homeState = CurrentState.ready;
+    homeState.value = CurrentState.busy;
+    pastQuizzes = await _graphQL.getQuizzesByTime(1);
+    activeQuizzes = await _graphQL.getQuizzesByTime(2);
+    upcomingQuizzes = await _graphQL.getQuizzesByTime(3);
+    desiredList = upcomingQuizzes;
+    homeState.value = CurrentState.ready;
     update();
   }
 
   @override
+  // ignore: unnecessary_overrides
```

Comment:
Why was this override needed? is onClose( ) compulsory?

Reply:
Yeah, it is kinda.